### PR TITLE
feat: stream support target databases

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

**Review by afwerar — APPROVED** [review]
LGTM

File: src/benchInsert.c [hunk 1]
```diff
@@ -217,6 +217,20 @@ int createDatabase(SDataBase* database) {
...
         return -1;
     }
+    if (g_arguments->taosc_version == 3) {
+        for (int i = 0; i < database->streams->size; i++) {
+            SSTREAM* stream = benchArrayGet(database->streams, i);
+            if (stream->drop) {
+                sprintf(command, "drop stream if exists %s", stream->stream_name);
+                infoPrint(stdout, "%s\n", command);
+            }
+            if (queryDbExec(conn, command)) {
```
Comment (sangshuduo) [inline src/benchInsert.c:227]: seems it's possible to call queryDbExec(conn, "\0")

File: src/benchInsert.c [hunk 2]
```diff
@@ -494,6 +508,10 @@ void postFreeResource() {
     for (int i = 0; i < g_arguments->databases->size; i++) {
         SDataBase * database = benchArrayGet(g_arguments->databases, i);
         benchArrayDestroy(database->cfgs);
+        for (int j = 0; j < database->streams->size; j++) {
+            SSTREAM* stream = benchArrayGet(database->streams, j);
+            tmfree(stream->target_dbs);
```
Comment (sangshuduo) [inline src/benchInsert.c:513]: Isn't stream NULL?

File: src/benchInsert.c [hunk 3]
```diff
@@ -1750,6 +1761,12 @@ int insertTestProcess() {
             SDataBase * database = benchArrayGet(g_arguments->databases, i);
             for (int j = 0; j < database->streams->size; ++j) {
                 SSTREAM * stream = benchArrayGet(database->streams, j);
+                for (int k = 0; k < stream->target_dbs->size; k++) {
+                    SDataBase* target_db = benchArrayGet(stream->target_dbs, k);
+                    if (createDatabase(target_db)) {
```
Comment (sangshuduo) [inline src/benchInsert.c:1766]: isn't target_db NULL?

File: src/benchJsonOpt.c [hunk 4]
```diff
@@ -777,7 +780,19 @@ static int getMetaFromInsertJsonFile(tools_cJSON *json) {
...
     for (int i = 0; i < dbSize; ++i) {
-        if (getDatabaseInfo(dbinfos, i)) {
+        SDataBase* database;
+        if (i > 0) {
+            database = benchCalloc(1, sizeof(SDataBase), true);
+            benchArrayPush(g_arguments->databases, database);
+            database->cfgs = benchArrayInit(1, sizeof(SDbCfg));
+        }
+        database = benchArrayGet(g_arguments->databases, i);
+        database->drop = true;
```
Comment (sangshuduo) [inline src/benchJsonOpt.c:790]: Isn't database NULL?

File: src/benchJsonOpt.c [hunk 5]
```diff
@@ -777,7 +780,19 @@ static int getMetaFromInsertJsonFile(tools_cJSON *json) {
     int dbSize = tools_cJSON_GetArraySize(dbinfos);
 
     for (int i = 0; i < dbSize; ++i) {
-        if (getDatabaseInfo(dbinfos, i)) {
+        SDataBase* database;
+        if (i > 0) {
+            database = benchCalloc(1, sizeof(SDataBase), true);
+            benchArrayPush(g_arguments->databases, database);
+            database->cfgs = benchArrayInit(1, sizeof(SDbCfg));
```
Comment (sangshuduo) [inline src/benchJsonOpt.c:787]: same question

**Review by sangshuduo — COMMENTED** [review]
see my comments

**Pull request closed without merging**